### PR TITLE
Web Demo Support running on Mac M1/2

### PR DESCRIPTION
**Pull request by cckuailong**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -174,6 +174,8 @@ python gradio_demo/app.py
 
 You could customize this script in [this file](gradio_demo/app.py).
 
+If you want to run it on MAC, you should follow [this Instruction](MacGPUEnv.md) and then rn the app.py.
```
Comment (EdwardOwusuAdjei) [inline README.md:178]: Typo, rn should be run.

Comment (cckuailong) [reply README.md]: Ahh, I made a mistake. Thank you bro.

Comment (EdwardOwusuAdjei) [reply README.md]: Amazing work! was waiting for this PR.

**Comment by EdwardOwusuAdjei** [comment]
For those on mac that used the requirement.txt in the project initially will now have to point to torchvision==0.16.2, a requirement by torch==2.1.2 and remove pytorch-cuda==11.8, or does not really matter to mention that?

**Comment by Paper99** [comment]
Could you update this PR according to the latest code? 
It's really helpful for us. And after that, we will merge it.

**Comment by cckuailong** [comment]
> @cckuailong @EdwardOwusuAdjei Could you update this PR according to the latest code? It's really helpful for us. And after that, we will merge it.

I have resolve the conflict and update the pr according to the latest code.

**Comment by OpenGG** [comment]
@cckuailong 

What are the RAM requirements? Is a M2 Pro with 32GB sufficient for running this project?

**Comment by cckuailong** [comment]
> @cckuailong
> 
> What are the RAM requirements? Is a M2 Pro with 32GB sufficient for running this project?

My test env is Mac Pro M1 32G.

**Comment by Paper99** [comment]
Thank you @cckuailong @OpenGG @EdwardOwusuAdjei. We've merged this PR.

**Pull request merged**